### DISCUSSION TITLE
DP node label selector for Upstream target

### DISCUSSION THREAD

**Discussion by shunsukew** [Kong/kong#12089] [Ideas and feature requests]
Within the Kong Gateway, the upstream functionality can have multiple targets, facilitating the distribution of traffic, as explained in the documentation available at https://docs.konghq.com/gateway/latest/key-concepts/upstreams.
In use-cases where DP nodes are geographically distributed or are deployed across multiple platforms such as cloud platforms (e.g. AWS, Azure, GCP) and baremetals in data centers, the backend target IPs or URLs may exhibit variations across regions or those platforms. Under such circumstances, the upstream configuration is apt for a specific region or platform but maybe wrong for others.

To address this, upstream target can have DP node label selector (akin to the label selector feature in Kubernetes) so that users can configure upstream target only for selected DP nodes (label could be region and cloudplatform names).

**Comment by StarlightIbuki** [comment]
Hi. Kong Mesh supports this usage.
A custom plugin can also achieve this to balance the incoming request dynamically.
